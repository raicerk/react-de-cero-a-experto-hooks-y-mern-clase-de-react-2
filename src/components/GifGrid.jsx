
import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);
    return (
        <>
            <h3>{category}</h3>
            {isLoading && (<h2>Loading...</h2>)}
            {
                images.map((image) => (<GifItem key={image.id} {...image} />))
            }
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}