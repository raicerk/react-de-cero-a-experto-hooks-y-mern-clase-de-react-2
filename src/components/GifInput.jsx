import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const GifInput = ({ onNewValue }) => {
    const InitialValue = ''
    const [input, setInput] = useState(InitialValue);

    const handlerChange = ({ target: { value } }) => {
        setInput(value)
    }

    const handlerSubmit = (event) => {
        event.preventDefault();
        if (input.trim().length <= 1) return;
        onNewValue(input.trim());
        setInput(InitialValue);
    }

    return (
        <form onSubmit={handlerSubmit} aria-label="form">
            <input
                type="text"
                value={input}
                onChange={handlerChange}
                placeholder="Busca tu gif acÃ¡ y presiona enter"
            />
        </form>
    )
}

GifInput.propTypes = {
    onNewValue: PropTypes.func.isRequired,
}