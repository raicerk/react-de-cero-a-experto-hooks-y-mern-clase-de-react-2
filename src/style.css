body{
    background-color: #1e1e1e;
    color: #fff;
}

input[type=text] {
    width: 15em;
    height: 2em;
}

input::placeholder {
    color: #1e1e1e
}

h3 {
    margin: 2em 0 2em 0;
    display: block;
}

div > p {
    color: #fff;
    font-size: 0.8em;
    text-transform: lowercase;
}
