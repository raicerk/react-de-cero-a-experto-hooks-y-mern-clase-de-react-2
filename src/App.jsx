import { useState } from 'react';
import { GifInput, GifGrid } from './components/';

export const App = () => {

    const [categories, setCategories] = useState(['Demon slayer'])

    const onNewValue = (value) => {
        if (categories.includes(value)) return;
        setCategories([value, ...categories])
    }

    return (
        <>
            <GifInput onNewValue={(value) => onNewValue(value)} />
            {
                categories.map((category) => (<GifGrid key={category} category={category} />))
            }
        </>
    )
}

export default App;