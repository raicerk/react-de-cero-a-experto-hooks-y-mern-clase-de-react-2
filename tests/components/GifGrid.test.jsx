import { render, screen } from '@testing-library/react';
import { GifGrid } from '../../src/components';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs')

describe('Test de <GifGrid/>', () => {

    const category = 'Naruto';
    
    test('Debe retornar el texto del titulo', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        })
        
        render(<GifGrid category={ category }/>)
        
        expect(screen.getByText('Naruto')).toBeTruthy();
        expect(screen.getByText('Loading...')).toBeTruthy();
        expect(screen.getByText('Naruto')).toMatchSnapshot();
    });

    
    
    test('Debe mostrar los elementos de la lista', () => {
        useFetchGifs.mockReturnValue({
            images: [
                {
                    id: '123',
                    title: 'Narutito',
                    url: 'https://raicerk.cl/naruto'
                }
            ],
            isLoading: false,
        })
        render(<GifGrid category={ category }/>)
        expect(screen.getAllByRole('img').length).toBe(1);
    })
    
})