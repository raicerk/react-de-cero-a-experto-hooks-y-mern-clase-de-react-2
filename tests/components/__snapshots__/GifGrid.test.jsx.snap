// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <GifGrid/> Debe retornar el texto del titulo 1`] = `
<h3>
  Naruto
</h3>
`;
